source/enemies/earthlyknight.js

```diff
@@ -0,0 +1,80 @@
+const { EnemyTemplate } = require("../classes/index.js");
+const { dealDamage, removeModifier, changeStagger } = require("../util/combatantUtil.js");
+const { isBuff } = require("../modifiers/_modifierDictionary");
+const { selectRandomFoe, selectNone, selectAllFoes } = require("../shared/actionComponents.js");
+const { spawnEnemy } = require("../util/roomUtil.js");
+const { getEmoji } = require("../util/elementUtil.js");
+
+const asteroid = require("./asteroid.js")
+
+const PATTERN = {
+  "Call Asteroid": "Tremor Smash",
+  "Damping Wallop": "random",
+  "Tremor Smash": "random"
+}
+function earthlyKnightPattern(actionName) {
+  return PATTERN[actionName]
+}
+
+module.exports = new EnemyTemplate("Earthly Knight",
+  "Earth",
+  250,
+  100,
+  "6",
+  0,
+  "Tremor Smash",
+  false
+).addAction({
+  name: "Damping Wallop",
+  element: "Earth",
+  description: `Inflict ${getEmoji("Earth")} damage and remove a buff`,
+  priority: 0,
+  effect: (targets, user, isCrit, adventure) => {
+    let damage = user.getPower() + 75;
+    changeStagger(targets, "elementMatchFoe");
+    let resultString = ""
+    for (target of targets) {
+      resultString += dealDamage([target], user, damage, false, user.element, adventure);
+      const targetBuffs = Object.keys(target.modifiers).filter(modifier => isBuff(modifier));
+      console.log(targetBuffs);
+      const buffIndex = adventure.generateRandomNumber(targetBuffs.length, "battle");
+      const rolledDebuff = targetBuffs[buffIndex];
+      const wasRemoved = removeModifier([target], { name: rolledDebuff, stacks: "all" }).length > 0;
+      if (wasRemoved) {
+        resultString += `${rolledDebuff} was lost.`;
```

Comment:
I think it's important to mention which target is losing their buffs. When the result is going to read XYXYXYX, where X is a damage statement and Y is a debuff loss statement (when some Ys might be missing), it'll be easy to accidentally pair the wrong Xs and Ys.

In that sense, I think the loop should be refactored to a `for(let i = 0; i < targets.length; i++)` so that the index can be used to both iterate over the targets for the debuff removal and find the target's name in the returned array of `getNames(targets, adventure)`.

I think it's an acceptable judgement call to consolidate the damage statements to all be at the beginning via `dealDamage(targets...`, but I'll leave that to you.